File: 백준/Silver/10828. 스택/스택.js
const fs = require('fs')

const [n, ...inputs] = fs.readFileSync('/dev/stdin').toString().trim().split("\n").map(s => s.trim());


const stk = []

for (let s of inputs) {
  const [a, b] = s.trim().split(" ");

  if (a === "push") {
    stk.push(b);
  } else if (a === "top") {
    console.log(stk.length === 0 ? -1 : stk.at(-1))
  } else if (a === "size") {
    console.log(stk.length)
  } else if (a === "empty") {
    console.log(stk.length === 0 ? 1 : 0)
  } else if (a === "pop") {
    console.log(stk.length === 0 ? -1 : stk.pop())
  }
}

File: 백준/Silver/10828. 스택/스택.py
import sys
input = sys.stdin.readline

stk = []

for _ in range(int(input())):
    s = input().strip()

    ar = s.split(" ")

    if ar[0] == "push":
        stk.append(ar[1])
        continue
    elif ar[0] == "pop":
        print(-1 if not stk else stk.pop())
        continue
    elif ar[0] == "size":
        print(len(stk))
        continue
    elif ar[0] == "empty":
        print(1 if not stk else 0)
        continue
    elif ar[0] == "top":
        print(-1 if not stk else stk[-1])
